### ISSUE TITLE
emcsfc_snow2mdl fails to build using GNU compiler

### ISSUE THREAD

**Issue by DusanJovic-NOAA** [NOAA-EMC/UFS_UTILS#554]
The compile error:
```
/home/dusan/simple-ufs/src/preproc/sorc/emcsfc_snow2mdl.fd/snow2mdl.F90:289:53:

  289 |                      lons_mdl(ij),lats_mdl(ij),nret)
      |                                                     1
Error: There is no specific subroutine for the generic ‘gdswzd’ at (1)                                                                            
```

**Comment by kgerheiser** [comment]
I have no problems building it with GNU 10.3.0.

**Comment by DusanJovic-NOAA** [comment]
Looks like I get the compile error when I use the develop branch of ip lib.

**Comment by kgerheiser** [comment]
Yes, ip is in a module now, and snow2mdl plays loose with type checking. I would recommend using tag `v3.3.3`. There's a branch of UFS_UTILS with the fix that will be merged.

**Comment by GeorgeGayno-NOAA** [comment]
UFS_UTILS uses ip v3.3.3 on all officially supported machines now. So can this issue be closed? Should I fix the snow2mdl code?

**Comment by DusanJovic-NOAA** [comment]
> UFS_UTILS uses ip v3.3.3 on all officially supported machines now. So can this issue be closed? Should I fix the snow2mdl code?

I can build UFS_UTILS using ip v3.3.3 with GNU. But related to building with GNU can you make this change:
```diff
$ git diff
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9c4e752f..3657c8ad 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -34,6 +34,9 @@ if(CMAKE_Fortran_COMPILER_ID MATCHES "^(Intel)$")
   set(CMAKE_Fortran_FLAGS_DEBUG "-O0 -check -check noarg_temp_created -check nopointer -fp-stack-check -fstack-protector-all -fpe0 -debug -ftrapuv")
 elseif(CMAKE_Fortran_COMPILER_ID MATCHES "^(GNU)$")
   set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -g -fbacktrace")
+  if(${CMAKE_Fortran_COMPILER_VERSION} VERSION_GREATER_EQUAL 10)
+    set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -fallow-argument-mismatch -fallow-invalid-boz")
+  endif()
   set(CMAKE_Fortran_FLAGS_RELEASE "-O3")
   set(CMAKE_Fortran_FLAGS_DEBUG "-O1 -ggdb -fno-unsafe-math-optimizations -frounding-math -fsignaling-nans -ffpe-trap=invalid,zero,overflow -fbounds-check -fsanitize=address -fno-omit-frame-pointer -fno-optimize-sibling-calls")
 endif()
```

**Comment by GeorgeGayno-NOAA** [comment]
> > UFS_UTILS uses ip v3.3.3 on all officially supported machines now. So can this issue be closed? Should I fix the snow2mdl code?
> 
> I can build UFS_UTILS using ip v3.3.3 with GNU. But related to building with GNU can you make this change:
> 
> ```diff
> $ git diff
> diff --git a/CMakeLists.txt b/CMakeLists.txt
> index 9c4e752f..3657c8ad 100644
> --- a/CMakeLists.txt
> +++ b/CMakeLists.txt
> @@ -34,6 +34,9 @@ if(CMAKE_Fortran_COMPILER_ID MATCHES "^(Intel)$")
>    set(CMAKE_Fortran_FLAGS_DEBUG "-O0 -check -check noarg_temp_created -check nopointer -fp-stack-check -fstack-protector-all -fpe0 -debug -ftrapuv")
>  elseif(CMAKE_Fortran_COMPILER_ID MATCHES "^(GNU)$")
>    set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -g -fbacktrace")
> +  if(${CMAKE_Fortran_COMPILER_VERSION} VERSION_GREATER_EQUAL 10)
> +    set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -fallow-argument-mismatch -fallow-invalid-boz")
> +  endif()
>    set(CMAKE_Fortran_FLAGS_RELEASE "-O3")
>    set(CMAKE_Fortran_FLAGS_DEBUG "-O1 -ggdb -fno-unsafe-math-optimizations -frounding-math -fsignaling-nans -ffpe-trap=invalid,zero,overflow -fbounds-check -fsanitize=address -fno-omit-frame-pointer -fno-optimize-sibling-calls")
>  endif()
> ```

Ok. I can work on this today.
